util/gas.ts

```diff
@@ -641,27 +641,32 @@ export const parseGasPrices = (common: Common, contents: string) => {
  * @param selectedFork The Hardfork selected by the user
  */
 export const findMatchingForkName = (
-  forks: Hardfork[],
+  forks: HardforkConfig[],
   forkNames: string[],
-  selectedFork: Hardfork | undefined,
+  selectedFork: HardforkConfig | undefined,
 ) => {
   // get all known forks mapped to a block number
   const knownForksWithBlocks = forks.reduce(
-    (res: { [forkName: string]: number }, fork: Hardfork) => {
+    (res: { [forkName: string]: number }, fork: HardforkConfig) => {
       if (fork.block) {
         res[fork.name] = fork.block
       }
+      // Hardcode "merge" block transition block's height as it's determined by TTD and not block height
+      if (fork.name === "merge") {
+        res[fork.name] = 15537394
+      }
       return res
     },
     {},
   )
 
   // filter all forks with block number below or equal to the selected,
   // sort in descending order and pick the first found
+  // Hardcode "merge" block transition block's height as it's determined by TTD and not block height
   let foundFork: string = forkNames
     .filter(
       (forkName) =>
-        knownForksWithBlocks[forkName] <= (selectedFork?.block || 0),
+        knownForksWithBlocks[forkName] <= (selectedFork?.block || selectedFork?.ttd ? 15537394 : 0 || 0),
```

Comment:
nit: The last || 0 is not needed as it will always default to 0 with this change